src/install/mod.rs

```diff
@@ -53,52 +54,53 @@ pub fn install_artifact(
 }
 
 fn tool_exists(tool_name: &str) -> Result<Option<Download>, failure::Error> {
-    if let Ok(path) = which(tool_name) {
+    if let Ok(Some((path, version))) = get_latest_installation(tool_name) {
         let no_parent_msg = format!("{} There is no path parent", emoji::WARN);
         log::debug!("found global {} binary at: {}", tool_name, path.display());
-        if !tool_needs_update(tool_name, &path)? {
-            return Ok(Some(Download::at(path.parent().expect(&no_parent_msg))));
+        if !tool_needs_update(tool_name, version)? {
+            Ok(Some(Download::at(path.parent().expect(&no_parent_msg))))
+        } else {
+            Ok(None)
         }
+    } else {
+        Ok(None)
     }
-
-    Ok(None)
 }
 
-fn tool_needs_update(tool_name: &str, path: &Path) -> Result<bool, failure::Error> {
-    let no_version_msg = format!("failed to find version for {}", tool_name);
-
-    let tool_version_output = Command::new(path.as_os_str())
-        .arg("--version")
-        .output()
-        .expect(&no_version_msg);
-
-    if !tool_version_output.status.success() {
-        let error = String::from_utf8_lossy(&tool_version_output.stderr);
-        log::debug!("could not find version for {}\n{}", tool_name, error);
-        return Ok(true);
+fn get_latest_installation(tool_name: &str) -> Result<Option<(PathBuf, Version)>, failure::Error> {
+    let mut result: Option<(PathBuf, Version)> = None;
+    for entry in fs::read_dir(&CACHE.destination)? {
+        let entry = entry?;
+        let filename = entry.file_name().into_string();
+        if let Ok(filename) = filename {
+            if filename.starts_with(tool_name) {
+                let version = filename
+                    .split(&format!("{}-v", tool_name))
+                    .collect::<Vec<&str>>()[1];
+                let version = Version::parse(version);
+                if let Ok(version) = version {
+                    let install = Some((entry.path(), version));
+                    // if let Some((_, latest_version)) = result {
+                    //     if version > latest_version {
+                    //         result = install;
+                    //     }
+                    // } else {
+                    //     result = install;
+                    // }
```

Comment:
this section is giving me trouble because 

```console
$ cargo check 
    Checking wrangler v1.6.0 (/Users/averyharnish/Documents/work/wrangler)
error[E0382]: use of moved value
  --> src/install/mod.rs:83:37
   |
83 |                     if let Some((_, latest_version)) = result {
   |                                     ^^^^^^^^^^^^^^ value moved here, in previous iteration of loop
   |
   = note: move occurs because value has type `semver::version::Version`, which does not implement the `Copy` trait

error[E0382]: borrow of moved value: `version`
  --> src/install/mod.rs:84:28
   |
81 |                 if let Ok(version) = version {
   |                           ------- move occurs because `version` has type `semver::version::Version`, which does not implement the `Copy` trait
82 |                     let install = Some((entry.path(), version));
   |                                                       ------- value moved here
83 |                     if let Some((_, latest_version)) = result {
84 |                         if version > latest_version {
   |                            ^^^^^^^ value borrowed here after move

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0382`.
error: could not compile `wrangler`.

To learn more, run the command again with --verbose.
```

Reply:
try 

```
83 |                     if let Some((_, latest_version)) = &result {

```

Reply:
i think i got it - only one clone now